contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -591,6 +591,33 @@ saved. For more details on how to use Merge Patch, see the RFC.
+| `//+listType=atomic/set/map` |  Applicable to objects that are of type list. If set to `atomic`, it defines that a single actor can replace the entire list, but cannot change individual elements of it. Unless set to `atomic`, different actors can update individual elements in the list. In that case, the listType can either be `set` (contains scalar elements only) or `map` (contains map-like elements). |
+| `//+listMapKeys=<list of keys>` |  Applicable in the context of lists being used as maps/dictionaries. It's a list of strings, that defines which combination of keys should be used as the identifier of each element.          |
+| `//+mapType=atomic/granular` |  If set to `atomic`, a single actor can only replace the map entirely. If set to `granular`, multiple actors can update individual map entries separately. |
+| `//+structType=atomic/granular` |  If set to `atomic`, a single actor can only replace the struct entirely. If set to `granular`, multiple actors can update individual struct fields separately. |
+
+##### Considerations on updates to patch strategies/topologies, and server-side apply
+
+A newer version of an API may decide to redefine the patch strategy or topology for a list, map, or struct. This section lists some scenarios and expected behaviours.
+
+* A `listType` can only be updated from `atomic` to `set` if the list elements are scalars. Similarly, `listType` updates from `atomic` to `map` will only succeed if the list items are key-value pairs, and a `listMapKeys` marker is added to define the list of keys that uniquely identify a list entry.
+* When updating `listType` from `set` to `atomic`, the next actor (e.g. `kubectl`) to apply the list will entirely replace it. The operation will not give out conflicts, even if the actor has't previously issued requests against this field.
+* When using a Kubernetes API version that does not support setting list, map and struct topologies (1.15 and before): lists, maps and structs are considered atomic. When upgrading to an API version that supports declaring topology for these fields (starting at 1.16):
+  * if `listType` is set to `set` or `map`
+  * if `listType` is set to `atomic`, `apply` operations using server-side apply will result in conflicts, unless they leave the field values unchanged.
+* Updates from specifying topologies to _not_ specifying them, will be equivalent to the topologies being set to `atomic`.
+* Updates between scalar and nested types are allowed, but are risky. If such a path must be followed, it's best to test outside a production system, and configure to `atomic` as a middle step.
+  * From `set` to `map`: while new objects can be created without problems following the updated topology, existing objects can no longer be changed.
+  * From `map` to `set`: there's high risk of losing or corrupting data of existing objects. Just reconfiguring `listType` without reapplying objects will empty the contents of existing objects.
```

Comment:
> Just reconfiguring `listType` without reapplying objects will empty the contents of existing objects.

@apelisse - Is this accurate? I'm not aware of any process that does this. Does this actually mean "empty contents" or does it mean "deduplicate items to form a set"?